src/ProjNet/IO/CoordinateSystems/CoordinateSystemWktReader.cs

```diff
@@ -358,53 +360,108 @@ private static ProjectedCoordinateSystem ReadProjectedCoordinateSystem(WktStream
             tokenizer.ReadToken("GEOGCS");
             var geographicCS = ReadGeographicCoordinateSystem(tokenizer);
             tokenizer.ReadToken(",");
-            IProjection projection = null;
-            IUnit unit = null;
-            var axes = new List<AxisInfo>(2);
+            tokenizer.ReadToken("PROJECTION");
```

Comment:
Requiring the input definition to be in a specific order is a breaking change, isn't it? Do we want that?

Reply:
From the OGC specs, I thought that the order was specific. I had changed this to get around this looping until the end of file. I'll rework so that it no longer needs to be in a specific order

Reply:
Hold on, maybe we get a second opinion.

Reply:
> second opinion

According to [the spec](https://portal.ogc.org/files/?artifact_id=999) section 7.2, it does appear that the standard only permits `PROJCS["<name>", <geographic cs>, <projection>` in that order (it's the projection parameters that might not have a defined order, at least as far as 7.2 is concerned).

That said, it's obviously difficult/impossible to rule out the idea that someone might be relying on this behavior.  I haven't verified one way or the other if this would be a behavior change, but if it is, then my opinion is that it would be **better** to avoid a behavior change if it's not too much work, but I wouldn't say it **needs** to keep supporting nonstandard things if that's an unreasonable amount of effort.

Reply:
Thinking about it further, the exception handling should be improved at the minimum, so I think I will rework so that it is agnostic to the order of the large-scale elements of the wkt string

Reply:
Maybe we can reduce this PR to the changes required to make `CompoundCoordinateSystem` and `VerticalCoordinateSystem` work and rework this function + exception handling in a different PR?

Reply:
It needs at least minor changes because currently when reading a compound coordinate system with the projected system as one of it's parts, without a change to this function, the reading will break. So one option would be to have it as it is now (which is following the specs) and then make it more flexible in the next one?

Reply:
Yeah, on (old) line 368, the while(ct != TokenType.Eol && ct != TokenType.Eof) means that it will read to the end of the string, not just to the end of the projected coordinate system block

Reply:
OK. I'll merge it then.